### ISSUE TITLE
Breadcrumb 30 characters limit and warning in production

### ISSUE THREAD

**Issue by cristianoccazinsp** [bugsnag/bugsnag-react-native#417]
After recently adding some screen tracking logic, I'm getting the following warning, which in turn also ends up on bugsnag (because I'm logging all warnings too!), which in turn will log again, and hopefully not end in a logging cycle.

```JSWarning · Breadcrumb name exceeds 30 characters (it has 31): Nav: Document1 -> NewInspection. It will be truncated.```

I understand the warning; however, it would be nice for warnings like this to not end up firing with production builds. Take for example RN's own warnings, they will always pollute your console, but they will not be included with release mode (not even the ones fired from native). 

Lastly, I'm wondering if the limit can be increased a bit. I've removed most unnecessary characters to remain below the limit, but there are cases it might go beyond it.

**Comment by mattdyoung** [comment]
Hi @cristianoccazinsp 

We do have plans to remove the 30 character breadcrumb name limit enforced by bugsnag-react-native, as we've recently improved the display of breadcrumbs in the Bugsnag dashboard which now allows longer names to be displayed. This is being done as part of a significant overhaul of the library to align the Javascript side with our bugsnag-js library.

For the time being you could specify a shorter description as the breadcrumb name and add any longer text as additional metadata on the breadcrumb:
https://docs.bugsnag.com/platforms/react-native/react-native/customizing-error-reports/#adding-detailed-breadcrumbs

E.g.
```
bugsnag.leaveBreadcrumb("Short breadcrumb name", {
  exampleField: "This text can be longer than 30 characters"
})
```

**Comment by cristianoccazinsp** [comment]
I've tried to reduce the breadcrumb length, but it really gets complicated to log previous/next screen with less than 30 chars.

![image](https://user-images.githubusercontent.com/48869228/67412565-e73d5780-f595-11e9-8081-b5d5e2d85d00.png)

It's interesting though, although there's a warning, the actual text doesn't seem trimmed.

Also, it is is really bad that the inernal JS code from bugsnag uses `console.warn`. There should be no console statements in any production build of any library.

**Comment by mattdyoung** [comment]
Hi @cristianoccazinsp 

The limit only applies to the breadcrumb name, not any fields within the breadcrumb.

So you shouldn't have to reduce the length of the text, just include it as a metadata field within the breadcrumb rather than as the name.

E.g. instead of:
```
bugsnag.leaveBreadcrumb("EditProperty > EditPropertyPreview")
```
use:
```
bugsnag.leaveBreadcrumb("Screen change", {
  screen: "EditProperty > EditPropertyPreview"
})
```

The breadcrumb name used to be truncated to 30 characters on our backend to ensure it could be displayed correctly. We've recently relaxed the limit on the backend and redesigned the UI to allow display of longer breadcrumb names, but we haven't yet completed the work to remove the limit where it was also enforced in our error reporting libraries. It looks like in the case of React Native, the limit wasn't actually enforced in the library but a warning was displayed that the backend would have truncated the name.

This warning is due to be removed in the next major version.

**Comment by cristianoccazinsp** [comment]
Thanks, that's very helpful!

**Comment by Nick-Lucas** [comment]
> It looks like in the case of React Native, the limit wasn't actually enforced in the library but a warning was displayed that the backend would have truncated the name. 
> This warning is due to be removed in the next major version.

Any update on this? Since the backend was already updated, and we can use longer names already, could you just remove the warning and push an update?

It seems a pretty quick win for the library

**Comment by mattdyoung** [comment]
@Nick-Lucas It's not just a case of removing the warning in the React Native library as there is still behaviour in the underlying iOS library (bugsnag-cocoa) which needs to be corrected in line with the changes too.

We're actively working on the next major versions of both so not planning to bring forward the fix at this time.

**Comment by Nick-Lucas** [comment]
Ah okay, thanks for the update on this. Looking forward to the next version. 😄

**Comment by johnkiely1** [comment]
This has now been fixed and is available in the latest version ([v7.3.4](https://github.com/bugsnag/bugsnag-js/releases/tag/v7.3.4)) of the Bugsnag react native library. As of v7.3 the library itself has now become part of the [bugsnag-js](https://github.com/bugsnag/bugsnag-js) repository. 
An upgrade guide can be found [here](https://github.com/bugsnag/bugsnag-js/blob/next/packages/react-native/UPGRADING.md)
With full documentation available [here](https://docs.bugsnag.com/platforms/react-native/react-native/).
